src/index.ts

```diff
@@ -143,7 +144,7 @@ export class Client {
     return this.transport
       .post<{
         sessionState: string;
-        methodResponses: [[IMethodName, ResponseType, string]];
+        methodResponses: ([IMethodName, ResponseType, string] | ['error', IError, string])[];
```

Comment:
OR 'error' could be considered as a method name, IError as a response type...

The underlying Datatype, to be cleaner than a Tuple, should be extracted to its own class and called `Invocation` to match RFC-8620. After a second thought here we really miss the Invocation concept that is central to RFC-8620...

Invocation is used regardless of regular execution or of error.

IMO given an Invocation, you could then be able to try to interprete it as a specific ResponseType and throw at that moment, no?

Reply:
I renamed the IMethodCall to IInvocation and improved the type, to use it in this method. Is it ok for you? I am not sure I can do better.